firmware/src/rdm.h

```diff
@@ -0,0 +1,315 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ * rdm.h
+ * Copyright (C) 2015 Simon Newton
+ */
+
+/**
+ * @defgroup rdm RDM
+ * @brief Remote Device Management
+ *
+ * @addtogroup rdm
+ * @{
+ * @file rdm.h
+ * @brief Remote Device Management
+ */
+
+#ifndef FIRMWARE_SRC_RDM_H_
+#define FIRMWARE_SRC_RDM_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+#include "transceiver.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief The root sub device.
+ */
+static const uint16_t SUBDEVICE_ROOT = 0;
+
+/**
+ * @brief All sub devices.
+ */
+static const uint16_t SUBDEVICE_ALL = 0xffff;
+
+/**
+ * @brief The maximum index for a sub device.
+ */
+static const uint16_t SUBDEVICE_MAX = 0x0200;
+
+/**
+ * @brief The RDM Sub Start Code.
+ */
+static const uint8_t SUB_START_CODE = 0x01;
+
+/**
+ * @brief The length of UIDs.
+ */
+enum {
+  UID_LENGTH = 6  //!< The size of a UID.
+};
+
+/**
+ * @brief Maximum size of RDM Parameter Data.
+ */
+enum {
+  MAX_PARAM_DATA_SIZE = 231  //!< Maximum size of RDM Parameter Data.
+};
+
+/**
+ * @brief RDM Command Classes from E1.20.
+ * @note See section 6.2.10 of ANSI E1.20 for more information.
+ */
+typedef enum {
+  DISCOVER_COMMAND = 0x10, /**< Discovery Command */
+  DISCOVER_COMMAND_RESPONSE = 0x11, /**< Discovery Response */
+  GET_COMMAND = 0x20, /**< Get Command */
+  GET_COMMAND_RESPONSE = 0x21, /**< Get Response */
+  SET_COMMAND = 0x30, /**< Set Command */
+  SET_COMMAND_RESPONSE = 0x31, /**< Set Response */
+} RDMCommandClass;
+
+/**
+ * @brief RDM Command Classes from E1.20.
+ * @note See section 6.2.10 of ANSI E1.20 for more information.
+ */
+typedef enum {
+  ACK = 0x00, /**< ACK */
+  ACK_TIMER = 0x01, /**< ACK Timer */
+  NACK_REASON = 0x02, /**< NACK with reason */
+  ACK_OVERFLOW = 0x03 /**< ACK OVERFLOW */
+} RDMResponseType;
+
+/**
+ * @brief RDM Parameter IDs.
+ * @note These come from E1.20, E1.37-1, E1.37-2 etc.
+ */
+typedef enum {
+  // Discovery
+  PID_DISC_UNIQUE_BRANCH = 0x0001,
+  PID_DISC_MUTE = 0x0002,
+  PID_DISC_UN_MUTE = 0x0003,
+  // network management
+  PID_PROXIED_DEVICES = 0x0010,
+  PID_PROXIED_DEVICE_COUNT = 0x0011,
+  PID_COMMS_STATUS = 0x0015,
+  // status collection
+  PID_QUEUED_MESSAGE = 0x0020,
+  PID_STATUS_MESSAGES = 0x0030,
+  PID_STATUS_ID_DESCRIPTION = 0x0031,
+  PID_CLEAR_STATUS_ID = 0x0032,
+  PID_SUB_DEVICE_STATUS_REPORT_THRESHOLD = 0x0033,
+  // RDM information
+  PID_SUPPORTED_PARAMETERS = 0x0050,
+  PID_PARAMETER_DESCRIPTION = 0x0051,
+  // production information
+  PID_DEVICE_INFO = 0x0060,
+  PID_PRODUCT_DETAIL_ID_LIST = 0x0070,
+  PID_DEVICE_MODEL_DESCRIPTION = 0x0080,
+  PID_MANUFACTURER_LABEL = 0x0081,
+  PID_DEVICE_LABEL = 0x0082,
+  PID_FACTORY_DEFAULTS = 0x0090,
+  PID_LANGUAGE_CAPABILITIES = 0x00A0,
+  PID_LANGUAGE = 0x00B0,
```

Comment:
Didn't we say lowercase for all of these? Then when you've fixed them you can apply the same thing to OLA. :)